linera-core/src/client/mod.rs

```diff
@@ -599,7 +599,7 @@ pub enum ChainClientError {
     CommitteeDeprecationError,
 
     #[error("Protocol error within chain client: {0}")]
-    ProtocolError(&'static str),
+    ProtocolError(String),
```

Comment:
Is this change needed for something in particular? It seems unrelated from the rest, and only for debugging/testing for that Protocol error issue we were getting, right?

In that case, to minimize divergence from upstream, I'd recommend reverting these changes in this file back